content/tutorials/how-to-send-l1-l2-transaction/10.index.md

```diff
@@ -0,0 +1,340 @@
+there are some common use cases that need L1 to L2 transaction functionality, such as:
+
+- Custom bridges.
+- Multi-layer governing smart contracts.
+
+## Requirements
+
+Before continue, make sure you have the following:
+
+- Destination contract ABI: in order to execute a transaction on a contract on L2,
+  you'll need the contract's ABI. For this example we'll assume a `Greeter.sol` contract is deployed
+  on ZKsync ([like this one](https://sepolia.explorer.zksync.io/address/0x543A5fBE705d040EFD63D9095054558FB4498F88)).
+- L1 RPC endpoint: to broadcast the L1->L2 transaction to the network.
+  You can find [public Sepolia RPC endpoints in Chainlist](https://chainlist.org/chain/11155111).
+- Account with ETH in L1 to pay the gas fees. Use any of the [faucets listed here](https://docs.zksync.io/ecosystem/network-faucets#sepolia-faucets).
+
+## Set up
+
+1. Create a project folder and `cd` into it
+
+    ```sh
+    mkdir cross-chain-tx
+    cd cross-chain-tx
+    ```
+
+1. Initialise the project:
+
+    ::code-group
+
+    ```bash [npm]
+    npm init -y
+    ```
+
+    ```bash [yarn]
+    yarn init -y
+    ```
+
+    ::
+
+1. Install the required dependencies:
+
+    ::code-group
+
+    ```bash [npm]
+    npm install -D zksync-ethers ethers typescript dotenv @matterlabs/zksync-contracts @types/node
+    ```
+
+    ```bash [yarn]
+    yarn add -D zksync-ethers ethers typescript dotenv @matterlabs/zksync-contracts @types/node
+
+    ```
+
+    ::
+
+1. Install `ts-node` globally to execute the scripts that we're going to create:
+
+    ::code-group
+
+    ```bash [npm]
+    npm install -g ts-node
+    ```
+
+    ```bash [yarn]
+    yarn global add ts-node
+
+    ```
+
+    ::
+
+1. In the root folder, add a `.env` file with the private key of the wallet to use:
+
+    ```md
+    WALLET_PRIVATE_KEY=0x..;
+    ```
+
+    ::callout{icon="i-heroicons-exclamation-triangle"}
+    Always use a separate wallet with no real funds for development.
+    Make sure your `.env` file is not pushed to an online repository by adding it to a `.gitignore` file.
+    ::
+
+## Step-by-step
+
+Sending an L1->L2 transaction requires following these steps:
+
+1. Initialise the providers and the wallet that will send the transaction:
+
+    ```ts
+    const l1provider = new Provider(L1_RPC_ENDPOINT);
+    const l2provider = new Provider(L2_RPC_ENDPOINT);
+    const wallet = new Wallet(WALLET_PRIVATE_KEY, l2provider, l1provider);
+    ```
+
+1. Retrieve the current gas price on L1.
+
+    ```ts
+    // retrieve L1 gas price
+    const l1GasPrice = await l1provider.getGasPrice();
+    console.log(`L1 gasPrice ${ethers.utils.formatEther(l1GasPrice)} ETH`);
+    ```
+
+1. Populate the transaction that you want to execute on L2:
+
+    ```ts
+    const contract = new Contract(L2_CONTRACT_ADDRESS, ABI, wallet);
+    // populate tx object
+    const tx = await contract.populateTransaction.setGreeting(message);
```

Comment:
Where is the `message` variable defined?

Reply:
will add it to a previous step where it's first used